### DISCUSSION TITLE
[client] createDeployment failing "MODULE_NOT_FOUND"

### DISCUSSION THREAD

**Discussion by eric-miller2129** [vercel/vercel#8466] [Help] [answered]
Hi there, 

I'm using `@vercel/client` to run a programmatic deployment to Vercel. Until recently it was working fine however it seems once I bumped the version from `12.1.11` to `12.2.0` it began to fail.

![Screen Shot 2022-08-25 at 17 38 57](https://user-images.githubusercontent.com/1424187/186773899-b9369942-df90-41a6-a486-80015b71c00d.png)

```
node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module 'tar-fs'
Require stack:
- .../node_modules/@vercel/client/dist/create-deployment.js
- .../node_modules/@vercel/client/dist/index.js
```
Switching back to `12.1.11` solves the issue, and just manually installing `tar-fs` also seems to solve the issue, but I don't think either of those things are the correct solution.

Could someone help me out?

Thanks!

**Comment by amyegan** [comment, marked as answer]
_The fix was released in [v12.2.3](https://www.npmjs.com/package/@vercel/client/v/12.2.3)_

Hi @eric-miller2129! Can you tell me more about what leads to this error? I'm most interested in knowing the specific command that throws this error. But any other information you can share about your app and deployment workflow may also help with debugging.

**Reply by eric-miller2129** [reply]
Hi @amyegan! Absolutely I can give you more information!

I am using the `@vercel/client` to create deployments on the fly. We have a team that uses our code to demo software to clients and as such we need to create different versions of them based off a set of parameters so we create unique vercel deployments for them.

This error message could definitely be a red herring however it seems to only show up on the latest `12.2.x`. The only command from Vercel that I use is `createDeployment()`

```
import { createDeployment } from '@vercel/client';

for await (const event of createDeployment({
        token: process.env.VERCEL_DEPLOY_TOKEN,
        path: (path: path.join(__dirname, '../../'),),
        teamId: 'teamId',
    }, 
    ...VercelDeploymentOptions
)) {
   ...
}
```

```
"dependencies": {
    "@vercel/build-utils": "^5.4.0",
    "@vercel/client": "^12.2.2",
}
```

As mentioned previously this worked just fine. Just running this code will fail before it makes it to the createDeployment method as far as I can tell.

**Reply by amyegan** [reply]
Thanks for sharing this! I was able to replicate the problem. A fix is on the way!

**Reply by eric-miller2129** [reply]
Thanks @amyegan!

**Reply by amyegan** [reply]
The fix was released yesterday in [v12.2.3](https://www.npmjs.com/package/@vercel/client/v/12.2.3)! Please let me know if you still see this problem, or any new ones, after updating 🙏
